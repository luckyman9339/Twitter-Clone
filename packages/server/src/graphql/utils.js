export const decodeCursor = cursor => {
  const content = new Buffer(cursor, 'base64').toString('binary');
  const [after, order] = content.split(';');
  return { after: new Date(after), order };
};

export const encodeCursor = ({ after, order }) => {
  const input = [new Date(after).toISOString(), order].join(';');
  return Buffer.from(input).toString('base64');
};

export const buildEdgesForTweetsPagination = (tweets, order) =>
  tweets.map(tweet => {
    const type = tweet.retweetForTweetId
      ? 'RETWEET'
      : tweet.likeForTweetId
      ? 'LIKE'
      : 'TWEET';

    const node = { ...tweet, type };

    return {
      cursor: encodeCursor({
        after: tweet.created_at,
        order,
      }),
      node,
    };
  });
