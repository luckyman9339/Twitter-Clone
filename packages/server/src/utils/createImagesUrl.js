module.exports = {
  createUserAvatarUrl: email => {
    return `https://api.adorable.io/avatars/200/${email}.png`;
  },
};
