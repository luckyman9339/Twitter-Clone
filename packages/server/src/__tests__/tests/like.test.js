import knexCleaner from 'knex-cleaner';
import { request } from '../utils';
import knex from 'db/knex';

let user = null;

beforeAll(async () => {
  await knexCleaner.clean(knex, {
    ignoreTables: ['knex_migrations', 'knex_migrations_lock'],
  });
  await knex.seed.run();
  user = (await knex.table('users').where({ username: 'test_user' }))[0];
});

it('should like a tweet', async () => {
  const context = { user };

  const query = `
    mutation LikeQuery($input: LikeInput!) {
      like(input: $input) {
        like {
          tweet {
            id
            content
            user {
              id
              username
            }
          }
          user {
            id
            username
          }
        }
      }
    }
  `;

  const variables = {
    input: { tweetId: 'e2a7c8d1-70c4-4e43-b7f0-a9c75d1f92ca' },
  };

  const result = await request(query, { context, variables });
  expect(result).toMatchSnapshot();
});
