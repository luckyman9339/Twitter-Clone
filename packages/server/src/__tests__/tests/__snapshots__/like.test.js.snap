// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should like a tweet 1`] = `
Object {
  "data": Object {
    "like": Object {
      "context": Object {
        "contextTweet": Object {
          "id": "cff2eee6-f27c-4991-a4cc-08996393615b",
        },
        "contextUser": Object {
          "id": "551cf7db-df68-4ad5-af6c-1935759b9291",
          "name": "Test User One",
        },
        "originalTweet": Object {
          "content": "Hello World from User Two!",
          "id": "e2a7c8d1-70c4-4e43-b7f0-a9c75d1f92ca",
        },
      },
    },
  },
}
`;
